wrappers/net/http/client.go

```diff
@@ -77,21 +77,32 @@ type TracingTransport struct {
 	// MetadataOnly flag overriding the configuration
 	MetadataOnly bool
 	tracer       tracer.Tracer
+	transport    http.RoundTripper
 }
 
 func NewTracingTransport(args ...context.Context) *TracingTransport {
+	return NewWrappedTracingTransport(http.DefaultTransport, args...)
+}
+
+func NewWrappedTracingTransport(rt http.RoundTripper, args ...context.Context) *TracingTransport {
 	currentTracer := internal.ExtractTracer(args)
 	return &TracingTransport{
-		tracer: currentTracer,
+		tracer:    currentTracer,
+		transport: rt,
 	}
 }
 
 // RoundTrip implements the RoundTripper interface to trace HTTP calls
 func (t *TracingTransport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	// if the TracingTransport is created before the global tracer is created it will be nil
+	if t.tracer == nil {
+		t.tracer = internal.ExtractTracer(nil)
```

Comment:
Can you add Epsagon debug in that scenario (using GlobalTracer)?

Reply:
To log a debug message that it defaulted to the global tracer?

Reply:
yes. (obviously, it will be possible only if the "new" t.tracer isn't nil as well)

Reply:
* added a test for the custom transport
* added a debug log when defaulting to the global tracer